scripts/sqlsmith/extract_failing_query.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# USAGE: ./extract_failing_query.sh [LOGPATH] [SQLPATH]
+# This script extracts failing query + its ddl + its dml from fuzzing log (LOGPATH),
+# writes out to sql script file (SQLPATH).
+#
+# EXAMPLE: ./extract_failing_query.sh [--pretty] fuzzing-sample.log fuzzing-golden.sql
+
+set -euo pipefail
+
+########## LIBRARY
+
+# Logs are of the form ... [EXECUTING XYZ]: <sql>
+# We strip the header, and append ';' to the sql.
+extract_sql() {
+    sed 's/.*\]: //' | sed 's/$/;/'
+}
+
+extract_sqlsmith_logs() {
+    grep "risingwave_sqlsmith::runner:" < "$1"
+}
+
+extract_ddl() {
+    grep "\[EXECUTING CREATE" | extract_sql
+}
+
+extract_dml() {
+    grep "\[EXECUTING POPULATION]" | extract_sql
+}
+
+# assumes it is last
+extract_query() {
+    tail -n 1 | extract_sql
+}
+
+########## EXTRACT QUERY + DDL + DML
+
+if [ "$1" = "--pretty" ]; then
```

Comment:
Will refactor and use `getopts` in a later PR. Planning to add basic shrinker here.